pkg/smb/controllerserver.go

```diff
@@ -311,15 +311,17 @@ func (d *Driver) smbVolToCSI(vol *smbVolume, parameters map[string]string) *csi.
 }
 
 // Given a CSI volume id, return a smbVolume
-func (d *Driver) getSmbVolFromID(id string) (*smbVolume, error) {
-	volRegex := regexp.MustCompile("^([^/]+)/([^/]+)$")
+// a sample volume Id: smb-server.default.svc.cluster.local/share/pvc-4729891a-f57e-4982-9c60-e9884af1be2f
+func getSmbVolFromID(id string) (*smbVolume, error) {
+	volRegex := regexp.MustCompile("^([^/]+)/(.*)/([^/]+)$")
```

Comment:
move this `volRegex` as a global var outside of this function.

Reply:
what if volume id is `smb-server.default.svc.cluster.local/share/dir1/dir2/dir3` ? would you also add a ut for this case?

Reply:
done.